src/transformStateWithClones.js

```
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        const newState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        currentState = newState;
+        break;
+    }
```

This switch lacks a default case. This violates checklist item 3: "3. [CODE STYLE]: switch/case should always have default case for error handling." 

Add a default branch to the switch to explicitly handle unknown action types (for example, throw an error like `throw new Error(`Unknown action type: ${action.type}`)`) so unexpected actions don't silently pass through.